src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -682,18 +682,11 @@ private EnumeratorResult GetEnumeratorInfo(ref ForEachEnumeratorInfo.Builder bui
                 }
             }
 
-            if ((object)collectionExprType == null) // There's no way to enumerate something without a type.
+            if (collectionExprType is null) // There's no way to enumerate something without a type.
             {
-                if (collectionExpr.Kind == BoundKind.DefaultExpression)
-                {
-                    diagnostics.Add(ErrorCode.ERR_DefaultLiteralNotValid, _syntax.Expression.Location);
-                }
-                else
-                {
-                    // The null and default literals were caught above, so anything else with a null type is a method group or anonymous function
-                    diagnostics.Add(ErrorCode.ERR_AnonMethGrpInForEach, _syntax.Expression.Location, collectionExpr.Display);
-                    // CONSIDER: dev10 also reports ERR_ForEachMissingMember (i.e. failed pattern match).
-                }
+                // The null and default literals were caught above, so anything else with a null type is a method group or anonymous function
```

Comment:
>and default literals were caught above [](start = 28, length = 38)

Is this still accurate?

Reply:
I think so. This is covered by `TestErrorDefaultLiteralCollection` (reports "no target type")

---
In reply to: [314115937](https://github.com/dotnet/roslyn/pull/37596#discussion_r314115937) [](ancestors = 314115937)

Reply:
> I think so. This is covered by TestErrorDefaultLiteralCollection (reports "no target type")

But it looks like it refers to the code that you removed. So, default literals are probably caught somewhere, but where exactly.

---
In reply to: [314130783](https://github.com/dotnet/roslyn/pull/37596#discussion_r314130783) [](ancestors = 314130783,314115937)